### PR TITLE
Add track duration

### PR DESCRIPTION
**Pull request by danielr18**
Solves #363.  Should support time strings and numbers as duration

### CODE REVIEW

File: packages/core/src/PlayerContextProvider.js [hunk 1]
```diff
@@ -80,12 +81,23 @@ const defaultState = {
...
   shouldForceLoad = false
 }) {
   const isNewTrack = prevState.activeTrackIndex !== index;
+  const currentTime = track.startingTime || 0;
+  let duration = 0;
+  if (Number.isFinite(track.duration)) {
+    duration = track.duration;
+  } else if (
+    typeof track.duration === 'string' ||
+    track.duration instanceof String
```
Comment (benwiley4000) [inline packages/core/src/PlayerContextProvider.js]: Any particular reason to support the `String` constructor? I don't think we're doing that anywhere else.

File: packages/core/src/PlayerContextProvider.js [hunk 2]
```diff
@@ -80,12 +81,23 @@ const defaultState = {
...
   index,
-  currentTime = 0,
+  track,
   shouldPlay = true,
   shouldForceLoad = false
 }) {
   const isNewTrack = prevState.activeTrackIndex !== index;
+  const currentTime = track.startingTime || 0;
+  let duration = 0;
+  if (Number.isFinite(track.duration)) {
```
[the remark this answers is not in the document]
Comment (benwiley4000) [reply packages/core/src/PlayerContextProvider.js]: What I'm imagining is something like:

```js
let duration = 0;
if (track.duration) {
  if (typeof track.duration === 'string') {
    duration = parseTimeString(track.duration);
  } else {
    duration = track.duration;
  }
}
```

[the remark this answers is not in the document]
Comment (danielr18) [reply packages/core/src/PlayerContextProvider.js]: Makes sense

**Comment by benwiley4000** [comment]
Can we also update the track proptype? https://github.com/danielr18/cassette/blob/5cd5b1c6a0644bbb105b52ffd26b75cebc00f7cf/packages/core/src/PlayerPropTypes.js#L76

**Comment by danielr18** [comment]
Yes, added it, I don't think duration needs to be required though.

**Comment by benwiley4000** [comment]
I'll follow your judgment. Thanks!

**Comment by benwiley4000** [comment]
Released in [v2.0.0-alpha.27](https://github.com/benwiley4000/cassette/releases/tag/v2.0.0-alpha.27) (has breaking changes, see the release notes)

**Pull request merged**